web/web.go

```diff
@@ -0,0 +1,771 @@
+		err = postgres.ProcessInfoSchema(conv, app.sourceDB)
+	default:
+		http.Error(w, fmt.Sprintf("Driver : '%s' is not supported", app.driver), http.StatusBadRequest)
+		return
+	}
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Schema Conversion Error : %v", err), http.StatusNotFound)
+		return
+	}
+	app.conv = conv
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(conv)
+}
+
+// Dump config is used for dump file approach.
+type DumpConfig struct {
+	Driver   string `json:"Driver"`
+	FilePath string `json:"Path"`
+}
+
+func convertSchemaDump(w http.ResponseWriter, r *http.Request) {
+	reqBody, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Body Read Error : %v", err), http.StatusInternalServerError)
+		return
+	}
+	var dc DumpConfig
+	err = json.Unmarshal(reqBody, &dc)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Request Body parse error : %v", err), http.StatusBadRequest)
+		return
+	}
+	f, err := os.Open(dc.FilePath)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to open the test data file: %v", err), http.StatusNotFound)
+		return
+	}
+	conv, err := conversion.SchemaConv(dc.Driver, &conversion.IOStreams{In: f, Out: os.Stdout}, 0)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Schema Conversion Error : %v", err), http.StatusNotFound)
+		return
+	}
+	app.conv = conv
+	app.driver = dc.Driver
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(conv)
+}
+
+// TODO: Add Index key statements.
+func getDDL(w http.ResponseWriter, r *http.Request) {
+	c := ddl.Config{Comments: true, ProtectIds: false}
+	var tables []string
+	for t := range app.conv.SpSchema {
+		tables = append(tables, t)
+	}
+	sort.Strings(tables)
+	ddl := make(map[string]string)
+	for _, t := range tables {
+		ddl[t] = app.conv.SpSchema[t].PrintCreateTable(c)
+	}
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(ddl)
+}
+
+func getSummary(w http.ResponseWriter, r *http.Request) {
+	reports := internal.AnalyzeTables(app.conv, nil)
+	summary := make(map[string]string)
+	for _, t := range reports {
+		var body string
+		for _, x := range t.Body {
+			body = body + x.Heading + "\n"
+			for i, l := range x.Lines {
+				body = body + fmt.Sprintf("%d) %s.\n", i+1, l) + "\n"
+			}
+		}
+		summary[t.SrcTable] = body
+	}
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(summary)
+}
+func writeHeading(s string) string {
+	return strings.Join([]string{
+		"----------------------------\n",
+		s, "\n",
+		"----------------------------\n"}, "")
+}
+func writeStmtStats(driverName string, conv *internal.Conv) string {
+	stmtstats := ""
+	type stat struct {
+		statement string
+		count     int64
+	}
+	var l []stat
+	for s, x := range conv.Stats.Statement {
+		l = append(l, stat{s, x.Schema + x.Data + x.Skip + x.Error})
+	}
+	// Sort by alphabetical order of statements.
+	sort.Slice(l, func(i, j int) bool {
+		return l[i].statement < l[j].statement
+	})
+	stmtstats = stmtstats + writeHeading("Statements Processed")
```

Comment:
In general sequences of appends to strings are very inefficient because strings are immutable in go, and so we're copying all of the time -- basically appending N strings to stmtstats becomes an O(N^2) operation.

As noted above, I suggest replace this body of code with a call to internal.writeStmtStats.
But if we didn't do that, then I'd probably suggest using a slice of strings, and then join them at the end to get the final string. Or use bytes.Buffer.

Reply:
Thanks, updated code as mentioned, using buffer